### PR TITLE
Add classmethods for overloaded Schedule constructor; add tests.

### PR DESCRIPTION
**Pull request by smallnamespace**
Implementation of overloaded `Schedule` constructors according to discussion at https://github.com/enthought/pyql/issues/112.

### CODE REVIEW

File: quantlib/time/schedule.pyx [hunk 1]
```diff
@@ -39,26 +42,82 @@ cdef public enum Rule:
...
+                deref(calendar._thisptr),
+                <_calendar.BusinessDayConvention>business_day_convention,
+                <_calendar.BusinessDayConvention>termination_date_convention,
+                <_schedule.Rule>date_generation_rule, <bool>end_of_month
+            )
+        return instance
 
     def __dealloc__(self):
-        if self._thisptr is NULL:
+        if self._thisptr is not NULL:
```
Comment (smallnamespace) [inline quantlib/time/schedule.pyx]: Can you please verify that `not` is correct here?

**Comment by smallnamespace** [comment]
Hm, the build failed because 1.4 only has a constructor that looks like this:
https://github.com/lballabio/QuantLib/blob/9ea2d7019a695994fc37a16e5ac35ceddfc2241b/ql/time/schedule.cpp#L64

How strict do we need to be on maintaining 1.4 compatibility?

**Comment by dpinte** [comment]
@smallnamespace we don't have to maintain compatibility with 1.4. The easiest is to track what ubuntu provides on the Travis-CI build vm's.

File: quantlib/time/schedule.pyx (hunk 1 above)
Comment (dpinte) [inline quantlib/time/schedule.pyx]: @smallnamespace that is fine!

**Comment by dpinte** [comment]
@smallnamespace LGTM except the fact that Travis is broken. I'll check if we can get a more recent ql built for Travis

**Comment by dpinte** [comment]
@smallnamespace this will fix the issue: https://github.com/enthought/pyql/pull/114.

Can you merge master in your branch once #114 is merged?

**Comment by smallnamespace** [comment]
Yes, I will get around to this sometime today and update the PR.

**Comment by smallnamespace** [comment]
@dpinte 

FYI, seems like 1.6 is the earliest QL version with the full 8-arg constructor.

**Comment by dpinte** [comment]
@smallnamespace I'll see if we can easily get QL `1.6`

**Comment by dpinte** [comment]
We now have support for QL 1.8. We need to merge master in this PR and re-run the tests

**Comment by dpinte** [comment]
Closing as superseded by #168

**Pull request closed without merging**